### PR TITLE
Convert to auto-property

### PR DESCRIPTION
**Pull request by gpetrou**

### CODE REVIEW

File: Src/Microsoft.Dynamic/Ast/LightLambdaExpression.cs [hunk 1]
```diff
@@ -27,35 +27,20 @@
...
-        private readonly IList<ParameterExpression> _args;
 
         internal LightLambdaExpression(Type retType, Expression body, string name, IList<ParameterExpression> args) {
-            _body = body;
-            _name = name;
-            _args = args;
+            Body = body;
+            Name = name;
+            Parameters = args;
             _retType = retType;
```
Comment (slozier) [inline Src/Microsoft.Dynamic/Ast/LightLambdaExpression.cs]: Any reason why you didn't convert ReturnType as well?

File: Src/Microsoft.Dynamic/Generation/TypeGen.cs [hunk 2]
```diff
@@ -38,9 +37,7 @@ public sealed class TypeGen {
             }
         }
 
-        internal AssemblyGen AssemblyGen {
-            get { return _myAssembly; }
-        }
+        internal AssemblyGen AssemblyGen { get; }
 
         public TypeBuilder TypeBuilder {
```
Comment (slozier) [inline Src/Microsoft.Dynamic/Generation/TypeGen.cs]: Why not this one?

File: Src/Microsoft.Dynamic/Hosting/Shell/ConsoleHostOptions.cs [hunk 3]
```diff
@@ -29,10 +29,9 @@ public enum Action {
             DisplayHelp
         }
 
-        private readonly List<string> _ignoredArgs = new List<string>();
         private readonly List<string> _environmentVars = new List<string>();
```
Comment (slozier) [inline Src/Microsoft.Dynamic/Hosting/Shell/ConsoleHostOptions.cs]: EnvironmentVariable?

**Comment by slide** [comment]
Waiting to hear back on @slozier's questions before merging.

**Pull request merged**